File: py-codellama7B-FP-all/taskid-147_get_max_triples-gen14.py
def get_max_triples(n: int) -> int:
    """
    You are given a positive integer n. You have to create an integer array a of length n.
        For each i (1 ≤ i ≤ n), the value of a[i] = i * i - i + 1.
        Return the number of triples (a[i], a[j], a[k]) of a where i < j < k, 
    and a[i] + a[j] + a[k] is a multiple of 3.

    Example :
    >>> get_max_triples(5)
    1
        Explanation: 
        a = [1, 3, 7, 13, 21]
        The only valid triple is (1, 7, 13).
    """

    # O(n log n)
    # def get_max_triples(n: int) -> int:
    #     a = list(range(1, n + 1))
    #     a = [i * i - i + 1 for i in a]
    #     a = sorted(a)
    #     count = 0
    #     for i in range(1, n):
    #         for j in range(i + 1, n):
    #             if a[i] + a[j] % 3 == 0:
    #                 count += 1
    #     return count

    # O(n)
    a = [i * i - i + 1 for i in range(1, n + 1)]
    count = 0
    for i in range(1, n):
        for j in range(i + 1, n):
            if a[i] + a[j] % 3 == 0:
                count += 1
    return count

File: java-codellama7B-AWQ-all/taskid-147_get_max_triples-gen14.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    // You are given a positive integer n. You have to create an integer array array list a of length n.
    // For each i (1 ≤ i ≤ n), the value of a[i] = i * i - i + 1.
    // Return the number of triples (a[i], a[j], a[k]) of a where i < j < k, 
    // and a[i] + a[j] + a[k] is a multiple of 3.
    // Example :
    // >>> getMaxTriples((5l))
    // (1l)
    // Explanation: 
    // a = [1, 3, 7, 13, 21]
    // The only valid triple is (1, 7, 13).
    public static long getMaxTriples(long n) {

        long counter = 0;
        for (long i = 1; i < n; i++) {
            for (long j = i + 1; j < n; j++) {
                for (long k = j + 1; k < n; k++) {
                    if (i * i - i + 1 + j * j - j + 1 + k * k - k + 1 % 3 == 0) {
                        counter++;
                    }
                }
            }
        }
        return counter;
    }
    public static void main(String[] args) {
    assert(getMaxTriples((5l)) == (1l));
    assert(getMaxTriples((6l)) == (4l));
    assert(getMaxTriples((10l)) == (36l));
    assert(getMaxTriples((100l)) == (53361l));
    }

}
